tchannel/tornado/hyperbahn.py

```diff
@@ -101,12 +101,20 @@ def _advertise_with_backoff(tchannel, service, timeout=None):
 
 
 @tornado.gen.coroutine
-def advertise(tchannel, service, routers, timeout=None):
+def advertise(tchannel, service, routers, timeout=None, router_file=None):
     """Advertise with Hyperbahn.
 
     See :py:class:`tchannel.TChannel.advertise`.
     """
     timeout = timeout or FIRST_ADVERTISE_TIME
+    if routers is None and router_file is not None:
+        # should just let the exceptions fly
+        with open(router_file, 'r') as json_data:
+            try:
+                routers = json.load(json_data)
+            except (IOError, OSError, ValueError):
```

Comment:
Same, probably worth a test case to try opening a non-existent file.

Actually @breerly how would you feel about only implementing this in the new TChannel API?